src/java/relex/RelationExtractor.java

```diff
@@ -380,27 +382,33 @@ public Sentence processSentence(String sentence)
 	TreeMap<String,Long> sumtime;
 	TreeMap<String,Long> cnttime;
 
+	private void startTime() {
+		if (logger.isDebugEnabled()) {
+			starttime = System.currentTimeMillis();
+		}
+	}
+
 	private void reportTime(String msg)
 	{
-		Long now = System.currentTimeMillis();
-		Long elapsed = now - starttime;
-		starttime = now;
+		if (logger.isDebugEnabled()) {
+			Long now = System.currentTimeMillis();
+			Long elapsed = now - starttime;
+			starttime = now;
+
+			Long sum = sumtime.get(msg);
+			Long cnt = cnttime.get(msg);
+			if (sum == null) {
```

Comment:
Since `sum` is an int, how could it ever be null? Are you telling me java mallocs a 64 bit pointer to store a 64-bit int?  I don't quite really believe it.

Reply:
Oh, I see, the old code did this too. Well, Hmmm.  Best leave old code alone, so that we don't re-expose some old bug.

Reply:
```Long``` in Java means that ```sum``` it is a reference to an instance of the Long class. It is common trick in Java using objects (```Long```) instead of primitives (```long```) to encode situation when value is not set. In latest Java versions Optional<T> can be used instead to make it explicit.